sources/settings-migrations/v1.29.0/kubernetes-device-ownership/src/main.rs

```diff
@@ -0,0 +1,20 @@
+use migration_helpers::common_migrations::AddSettingsMigration;
+use migration_helpers::{migrate, Result};
+use std::process;
+
+///  We added a new setting, `settings.kubernetes.device-ownership-from-security-context`
+fn run() -> Result<()> {
+    migrate(AddSettingsMigration(&[
+        "settings.kubernetes.device-ownership-from-security-context",
+    ]))
```

Comment:
You will also want to add migration for the affected services so that the containerd service gets restarted upon migration. 

Reference this code - https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/settings-migrations/v1.23.0/nvidia-container-runtime-metadata/src/main.rs#L7-L10

Reply:
That makes sense.

Should the Metadata Migration be in addition to the existing `SettingsMigration` or in place of it? 

From my read of the migration logic for [`AddMetadataMigration`](https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/api/migration/migration-helpers/src/common_migrations.rs#L1183) it looks like it only removes the `affected-services` and not the actual setting from the datastore.

Reply:
Discussed offline. Reference PR #4182  where both a settings and a metadata migrations were added to support a new setting